pykoi/rlhf/supervised_finetuning.py

```diff
@@ -70,26 +81,40 @@ def __init__(self, rlhf_config: RLHFConfig, enable_telemetry: bool = True) -> No
             gradient_accumulation_steps=self._rlhf_config.gradient_accumulation_steps,
             gradient_checkpointing=self._rlhf_config.gradient_checkpointing,
             gradient_checkpointing_kwargs={
-                "use_reentrant": self._rlhf_config.gradient_checkpointing_use_reentrant
-            },
+                "use_reentrant": self._rlhf_config.gradient_checkpointing_use_reentrant},
             fp16=self._rlhf_config.fp16,
             bf16=self._rlhf_config.bf16,
             weight_decay=self._rlhf_config.weight_decay,
             run_name="step1_supervised_finetuning",
             ddp_find_unused_parameters=False,
+            num_train_epochs=self._rlhf_config.num_train_epochs,
         )
         self.model = AutoModelForCausalLM.from_pretrained(
             self._rlhf_config.base_model_path,
             load_in_8bit=self._rlhf_config.load_in_8bit,
             device_map=self._rlhf_config.device_map,
         )
+        # resize the token embeddings to include the added special tokens
+        self.model.resize_token_embeddings(len(self.tokenizer))
+        data_collator = None
+        if self._rlhf_config.data_collator == "DataCollatorForCompletionOnlyLM":
```

Comment:
You should consider to set `data_collator` to `DataCollatorForCompletionOnlyLM` class instead of a string for SFT training argument.

Then, here you should check `None`. Also, it looks like a bug for me that if people use SFT without passing in data_collator. Therefore, you should set proper default value in the config.py

Reply:
I believe if set to None, the default Datacollator will be used when "None" is passed to trl.SFTTrainer.